contracts/aggregator/common/main.sol

```diff
@@ -0,0 +1,204 @@
+        uint256 length = routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            if (routes[i] == _route) {
+                delete routes[i];
+                delete routeToImpl[_route];
+                delete routeStatus[_route];
+            }
+        }
+    }
+
+    /**
+     * @dev Function to transfer fee to the treasury.
+     * @notice Function to transfer fee to the treasury. Will be called manually.
+     * @param _tokens token addresses for transferring fee to treasury.
+     * @param _to treasury address.
+     */
+    function transferFee(address[] memory _tokens, address _to)
+        public
+        onlyOwner
+    {
+        uint256[] memory _amts = new uint256[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            IERC20 token_ = IERC20(_tokens[i]);
+            uint256 decimals_ = TokenInterface(_tokens[i]).decimals();
+            uint256 amtToSub_ = decimals_ == 18 ? 1e10 : decimals_ > 12
+                ? 10000
+                : decimals_ > 7
+                ? 100
+                : 10;
+            _amts[i] = token_.balanceOf(address(this)) > amtToSub_
+                ? (token_.balanceOf(address(this)) - amtToSub_)
+                : 0;
+            if (_amts[i] > 0) token_.safeTransfer(_to, _amts[i]);
+        }
+        emit LogCollectRevenue(_to, _tokens, _amts);
+    }
+}
+
+contract FlashAggregator is AdminModule {
+    /**
+     * @dev Returns fee for the passed route in BPS.
+     * @notice Returns fee for the passed route in BPS. 1 BPS == 0.01%.
+     * @param _route route number for flashloan.
+     */
+    function calculateFeeBPS(uint256 _route)
+        public
+        view
+        returns (uint256 BPS_)
+    {
+        bytes memory _output = Address.functionStaticCall(
+            routeToImpl[_route],
+            msg.data,
+            "calculateFeeBPS-call-failed"
+        );
+        BPS_ = abi.decode(_output, (uint256));
+
+        if (BPS_ < InstaFeeBPS) {
+            BPS_ = InstaFeeBPS;
+        }
+    }
+
+    /**
+     * @dev Function to get the list of all routes.
+     * @notice Function to get the list of all routes.
+     */
+    function getRoutes() public view returns (uint256[] memory) {
+        return routes;
+    }
+
+    /**
+     * @dev Function to get the list of enabled routes.
+     * @notice Function to get the list of enabled routes.
+     */
+    function getEnabledRoutes()
+        public
+        view
+        returns (uint16[] memory routesEnabled_)
+    {
+        uint256[] memory routesAll_ = getRoutes();
+        uint256 length = routesAll_.length;
+        uint256 _count = 0;
+
+        for (uint256 i = 0; i < length; i++) {
+            if (routeStatus[routesAll_[i]] == true) {
+                _count++;
+            }
+        }
+
+        for (uint256 j = 0; j < _count; j++) {
+            if (routeStatus[routesAll_[j]]) {
+                routesEnabled_[j] = uint16(routesAll_[j]);
+            }
+        }
+    }
```

Comment:
@shriyatyagii What is this function doing? Get the `count` of enabled routes and loop the first `count` elements of the array and put the ones which are enabled to the list? Also where have we defined the length of the `routesEnabled_` array. Have you tested this function?

Reply:
```suggestion
    function getEnabledRoutes()
        public
        view
        returns (uint16[] memory routesEnabled_)
    {
        uint256[] memory routesAll_ = getRoutes();
        uint256 length = routesAll_.length;
        uint256 _count = 0;

        for (uint256 i = 0; i < length; i++) {
            if (routeStatus[routesAll_[i]] == true) {
                _count++;
            }
        }
        routesEnabled_ = new uint16[](_count);
        uint256 j = 0;
        for (uint256 i = 0; i < length; i++) {
            if (routeStatus[routesAll_[i]]) {
                routesEnabled_[j] = uint16(routesAll_[i]);
                j++;
            }
        }
    }
```

Reply:
No, didn't test this last night. Fixed this!